libutils/src/globalattributes.cpp

```diff
@@ -20,156 +20,205 @@
-
-	PgModelerAppPath=getPathFromEnv(QString("PGMODELER_APP_PATH"),
-									  QString(BINDIR) + QString("/pgmodeler"),
-									  QString("./pgmodeler"));
-
+	PgModelerCLIPath(getPathFromEnv(
+		QString("PGMODELER_CLI_PATH"),
+		QString(BINDIR)				+ QString("/pgmodeler-cli"),
+		app.applicationDirPath()	+ QString("/pgmodeler-cli"))),
 
+	PgModelerAppPath(getPathFromEnv(
+		QString("PGMODELER_APP_PATH"),
+		QString(BINDIR)				+ QString("/pgmodeler"),
+		app.applicationDirPath()	+ QString("/pgmodeler")))
 #else
-	PgModelerCHandlerPath=getPathFromEnv(QString("PGMODELER_CHANDLER_PATH"),
-										   QString(PRIVATEBINDIR) + QString("\\pgmodeler-ch.exe"),
-										   QString(".\\pgmodeler-ch.exe")),
-
-	PgModelerCLIPath=getPathFromEnv(QString("PGMODELER_CLI_PATH"),
-									  QString(PRIVATEBINDIR) + QString("\\pgmodeler-cli.exe"),
-									  QString(".\\pgmodeler-cli.exe")),
-
-	PgModelerAppPath=getPathFromEnv(QString("PGMODELER_APP_PATH"),
-									  QString(BINDIR) + QString("\\pgmodeler.exe"),
-									  QString(".\\pgmodeler.exe"));
-
+	PgModelerCHandlerPath(getPathFromEnv(
+		QString("PGMODELER_CHANDLER_PATH"),
+		QString(PRIVATEBINDIR)		+ QString("\\pgmodeler-ch.exe"),
+		app.applicationDirPath()	+ QString("\\pgmodeler-ch.exe"))),
+
+	PgModelerCLIPath(getPathFromEnv(
+		QString("PGMODELER_CLI_PATH"),
+		QString(PRIVATEBINDIR)		+ QString("\\pgmodeler-cli.exe"),
+		app.applicationDirPath()	+ QString("\\pgmodeler-cli.exe"))),
+
+	PgModelerAppPath(getPathFromEnv(
+		QString("PGMODELER_APP_PATH"),
+		QString(BINDIR)				+ QString("\\pgmodeler.exe"),
+		app.applicationDirPath()	+ QString("\\pgmodeler.exe")))
 #endif
 
 #ifdef DEMO_VERSION
 	//Maximum object creation counter for demo version
-	const unsigned MaxObjectCount=15;
+	, MaxObjectCount(15)
 #endif
+{
+}
 
+QString GlobalAttributes::getPathFromEnv(const QString &varname, const QString &default_val, const QString &fallback_val)
+{
+	QFileInfo fi;
+	QStringList paths={ QDir::toNativeSeparators(getenv(varname.toStdString().c_str())),
+						QDir::toNativeSeparators(default_val) };
 
-	QString getPathFromEnv(const QString &varname, const QString &default_val, const QString &fallback_val)
+	for(int i=0; i < 2; i++)
 	{
-		QFileInfo fi;
-		QStringList paths={ QDir::toNativeSeparators(getenv(varname.toStdString().c_str())),
-							QDir::toNativeSeparators(default_val) };
-
-		for(int i=0; i < 2; i++)
-		{
-			fi.setFile(paths[i]);
-			if(fi.exists() || (i==1 && fallback_val.isEmpty()))
-				return(paths[i].replace('\\','/'));
-		}
-
-		fi.setFile(fallback_val);
-		return(fi.absoluteFilePath());
+		fi.setFile(paths[i]);
+		if(fi.exists() || (i==1 && fallback_val.isEmpty()))
+			return(paths[i].replace('\\','/'));
 	}
+
+	fi.setFile(fallback_val);
+	return(fi.absoluteFilePath());
+}
+
+GlobalAttributes& GlobalAttributes::create(const QApplication& app)
+{
+	GlobalAttributes::get()._instance = new GlobalAttributes(app);
+
+	return *GlobalAttributes::get()._instance;
```

Comment:
This is why it crashes. It goes in an endless loop.
Should be :
return *GlobalAttributes::get()._instance;